tests/functional/conftest.py

```diff
@@ -90,3 +92,41 @@ def _run_get_target_info(target_path):
         return updater.get_targetinfo(target_path)
 
     return _run_get_target_info
+
+
+@pytest.fixture
+def task_completed_within_threshold():
+    def _run_task_completion_check(
+        http_request, headers, response_json, threshold
+    ):
+        """Validates that a task has finished within a threshold of seconds."""
+        task_id = response_json["data"]["task_id"]
+        task_submitted = dateutil.parser.parse(
+            response_json["data"]["last_update"]
+        )
+        while (
+            (datetime.utcnow() - task_submitted).total_seconds()
+        ) <= threshold:
+            response = http_request(
+                method="GET",
+                url=f"/api/v1/task/?task_id={task_id}",
+                headers=headers,
+            )
+            assert response.status_code == 200, response.text
+            rp_json = response.json()
+            # "result" is missing when the task is still "PENDING"
+            status = ""
+            result = rp_json["data"].get("result")
+            if result is not None:
+                # "status" is missing when the task is in its earliest stage
+                status = result.get("status")
+
+            if status == "Task finished.":
+                break
+
+        if ((datetime.utcnow() - task_submitted).total_seconds()) > threshold:
```

Comment:
You don't need double
```suggestion
        if datetime.utcnow() - task_submitted).total_seconds() > threshold:
```

Reply:
I tried it locally and it didn't work: 
![Untitled](https://s3-us-west-2.amazonaws.com/secure.notion-static.com/ae77b226-ea0e-41ef-940f-65e6113add7f/Untitled.png)

I need to have timedelta object in order to work: 
<img width="291" alt="image" src="https://user-images.githubusercontent.com/16246778/209151803-37073ed3-02a4-4fc0-addf-8b58c0cbd43f.png">

Then, because of the order of operations, I will have to use brackets in order to do the subtraction between `datetime.utcnow() - task_submitted` before calling `total_seconds()` on the resulting timedelta object.

Reply:
Sorry. My fault.